xDrip/Extensions/TimeFrameSelectionView.swift

```diff
@@ -0,0 +1,176 @@
+//
+//  TimeFrameSelectionView.swift
+//  xDrip
+//
+//  Created by Ivan Skoryk on 20.03.2020.
+//  Copyright © 2020 Faifly. All rights reserved.
+//
+
+import UIKit
+
+protocol TimeFrameSelectionDelegate: class {
+    func segmentDidChange(index: Int)
+}
+
+class TimeFrameSelectionView: UIView {
+    
+    @IBOutlet var contentView: UIView!
+    @IBOutlet weak var selectedSegmentBackgroundView: UIView!
+    @IBOutlet weak var selectedSegmentViewWidthConstraint: NSLayoutConstraint!
+    @IBOutlet weak var selectedSegmentViewLeadingConstraint: NSLayoutConstraint!
+    
+    weak var delegate: TimeFrameSelectionDelegate?
+    
+    private var separators: [UIView] = []
+    private var stackView = UIStackView()
+    private var selectedSegment = 0
+    
+    private var hiddenSeparators: [UIView?] = []
+    
+    private var singleSegmentWidth: CGFloat = 0.0
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        self.commonInit()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        self.commonInit()
+    }
+    
+    private func commonInit() {
+        Bundle.main.loadNibNamed("TimeFrameSelectionView", owner: self, options: nil)
+
+        addSubview(contentView)
+        contentView.frame = self.bounds
+    }
+    
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        
+        self.contentView.layer.cornerRadius = 9
+        self.selectedSegmentBackgroundView.layer.cornerRadius = 7
+        
+        self.selectedSegmentBackgroundView.layer.shadowOffset = CGSize(width: 0, height: 3)
+        self.selectedSegmentBackgroundView.layer.shadowRadius = 3
+        self.selectedSegmentBackgroundView.layer.shadowOpacity = 0.12
+        self.selectedSegmentBackgroundView.layer.shadowColor = UIColor.black.cgColor
+    }
+    
+    func config(with buttons: [String]) {
+        if stackView.superview != nil {
+            stackView.removeFromSuperview()
+        }
+        
+        separators.forEach { (view) in
+            view.removeFromSuperview()
+        }
+        
+        self.singleSegmentWidth = (self.contentView.frame.width - CGFloat(4 * buttons.count)) / CGFloat(buttons.count)
```

Comment:
The width is set statically single time and the view isn't correctly resized, especially for macOS case.